src/openCypherParser/AST/CypherVisitor.cs

```diff
@@ -530,6 +530,66 @@ private void UpdateEntityTypesInExpressionInPlace(IEnumerable<QueryExpression> e
             });
         }
 
+        // the order of parsing types referred from following:
+        // https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/integral-numeric-types
+        QueryExpressionValue ParseIntegerLiteral(string literal)
+        {
+            // sbyte was excluded due to no equivalent type in sql type mapping
+            if (byte.TryParse(literal, out byte byteValue))
+            {
+                return new QueryExpressionValue()
+                {
+                    Value = byteValue
+                };
+            }
+            else if (short.TryParse(literal, out short shortValue))
+            {
+                return new QueryExpressionValue()
+                {
+                    Value = shortValue
+                };
+            }
+            else if (ushort.TryParse(literal, out ushort ushortValue))
+            {
+                return new QueryExpressionValue()
+                {
+                    Value = ushortValue
+                };
+            }
+            else if (int.TryParse(literal, out int intValue))
+            {
+                return new QueryExpressionValue()
+                {
+                    Value = intValue
+                };
+            }
+            else if (uint.TryParse(literal, out uint uintValue))
+            {
+                return new QueryExpressionValue()
+                {
+                    Value = uintValue
+                };
+            }
+            else if (long.TryParse(literal, out long longValue))
+            {
+                return new QueryExpressionValue()
+                {
+                    Value = longValue
+                };
+            }
+            else if (ulong.TryParse(literal, out ulong ulongValue))
+            {
+                return new QueryExpressionValue()
+                {
+                    Value = ulongValue
+                };
+            }
+            else
+            {
+                throw new TranspilerNotSupportedException($"Integer literal :{literal} out of supported range");
```

Comment:
I think in the exception message, we append 'is not supported'. so you only need to specific 'what' is not supported in the message parameter.

Reply:
"out of supported range" was removed. Thanks!